src/app/components/shared/datetime/abstract-datetime.component.ts

```diff
@@ -0,0 +1,28 @@
+export interface IAbstractDatetimeComponent {
+  formattedValue: () => string;
+  tooltipValue: () => string;
+  rawDateTime: () => string;
+  suffix: () => string;
+}
+
+/**
+ * An abstract component that should be used in conjunction with the abstract-datetime.component.html template
+ *
+ * @requires formattedValue
+ * @requires tooltipValue
+ * @requires rawDateTime
+ */
+export abstract class AbstractDatetimeComponent
+  implements IAbstractDatetimeComponent
+{
+  public constructor() {}
+
+  public abstract formattedValue(): string;
+  public abstract tooltipValue(): string;
+  public abstract rawDateTime(): string;
+
+  // I have implemented this method so that each datetime component will default to having no suffix
+  public suffix(): string {
+    return "";
+  }
```

Comment:
i'm going to suggest redefining the shape of this class.

```suggestion
  public formattedValue: string;
  public tooltipValue: string;
  public rawDateTime: string = "";

 public abstract update() { };
```